homeassistant/components/modbus/sensor.py

```diff
@@ -220,9 +220,7 @@ def __init__(
 
     async def async_added_to_hass(self):
         """Handle entity which will be added."""
-        state = await self.async_get_last_state()
-        if state:
-            self._value = state.state
+        self._value = await self.async_get_last_state()
```

Comment:
`async_get_last_state` returns a `State` instance or `None`.

https://github.com/home-assistant/core/blob/49c23bad29785fd541f9c2e0c09ae3dad795f22a/homeassistant/helpers/restore_state.py#L260

Reply:
you are right, how can I (in pytest) simulate a None. In pytest it always return a state?

Reply:
Removed it for now. I think we (I) should make a general pytest helper for testing restore of states in a platform.

Reply:
Yeah, I haven't looked into this very much how (if) we normally test this.